pandas/_libs/tslib.pyx

```diff
@@ -700,15 +700,15 @@ def array_to_datetime_with_tz(ndarray values, tzinfo tz, NPY_DATETIMEUNIT creso)
             ival = NPY_NAT
 
         else:
-            ts = Timestamp(item)
+            if PyDateTime_Check(item) and item.tzinfo is not None:
+                # We can't call Timestamp constructor with a tz arg, have to
+                #  do 2-step
+                ts = Timestamp(item).tz_convert(tz)
+            else:
+                ts = Timestamp(item, tz=tz)
```

Comment:
Nit: Should we use `tz_localize` here to be more explicit?

Reply:
the point is to match the behavior of calling the constructor this way, so i prefer this.

But now that #55712 is merged we can replace all of this with `convert_to_tsobject` which will move the ball down the field on a bunch of fronts.  will update

Reply:
looks like moving to convert_to_tsobject introduces its own set of issues, so ill do that in a separate PR